src/sdl2/messagebox.rs

```diff
@@ -48,3 +101,60 @@ pub fn show_simple_message_box(flags: MessageBoxFlag, title: &str,
         Err(SdlError(get_error()))
     }
 }
+
+/// returns the id of the button triggered
+pub fn show_message_box<'a>(flags:MessageBoxFlag, buttons:&'a [ButtonData], title:&str,
```

Comment:
I'm fine with both, I just thought that integers would be easier to handle than a potential reference of ButtonData. I think it can lead to lifetime problems too, but I'm not sure. Either way is fine by me.

I think it would be better to stay close to the raw SDL implementation and keep the Result<i32>, but that's just my opinion.